salt/modules/zypperpkg.py

```diff
@@ -1700,28 +1733,35 @@ def upgrade(refresh=True,
             cmd_update.extend(['--from' if dist_upgrade else '--repo', repo])
         log.info('Targeting repos: %s', fromrepo)
 
-    if dist_upgrade:
-        # TODO: Grains validation should be moved to Zypper class
-        if __grains__["osrelease_info"][0] > 11:
-            if novendorchange:
-                cmd_update.append("--no-allow-vendor-change")
-                log.info("Disabling vendor changes")
-            else:
+    # TODO: Grains validation should be moved to Zypper class
+    if not novendorchange:
+        __zypper__(root=root).calc_zypper()
+        if dist_upgrade:
+            if __zypper__(root=root).inst_avc or __zypper__(root=root).dup_avc:
```

Comment:
In case you fixed the inconsistency with the `inst_avc` and `dup_avc` flags I mentioned above, I think this can be changed to:

```suggestion
            if __zypper__(root=root).dup_avc:
```